devices/stm32f429xx/stm32f429xx_usart.c

```diff
@@ -0,0 +1,597 @@
+/* Copyright 2015 Adam Green (http://mbed.org/users/AdamGreen/)
+   Copyright 2015 JaredCJR   (https://github.com/JaredCJR)
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License as published
+   by the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/* Routines used to provide STM32F429xx USART functionality to the mri debugger. */
+#include <string.h>
+#include <stdlib.h>
+#include "platforms.h"
+#include "../../architectures/armv7-m/debug_cm3.h"
+#include "stm32f429xx_init.h"
+#include "stm32f429xx_usart.h"
+
+//start count by 0,1,2,etc
+static const UartConfiguration g_uartConfigurations[] = {
+    {
+		/* 
+		 * Tx=PA9
+		 * Rx=PA10
+		 */
+        USART1,
+        7,//AF7
+        7//AF7
+    },
+    {
+		/* 
+		 * Tx=PD5
+		 * Rx=PD6
+		 */
+        USART2,
+        7,//AF7
+        7//AF7
+    },
+    {
+		/* 
+		 * Tx=PB10
+		 * Rx=PB11
+		 */
+        USART3,
+        7,//AF7
+        7//AF7
+    }
+};
+
+
+
+typedef struct {
+    int      share;
+    uint32_t uartIndex;
+    uint32_t baudRate;
+} UartParameters;
+
+
+static void 	configureNVICForUartInterrupt(uint32_t index);
+static void     parseUartParameters(Token* pParameterTokens, UartParameters* pParameters);
+static void     setManualBaudFlag(void);
+static void     setUartSharedFlag(void);
+static void     saveUartToBeUsedByDebugger(uint32_t mriUart);
+static void  	configureUartForExclusiveUseOfDebugger(UartParameters* pParameters);
+
+void __mriStm32f429xxUart_Init(Token *pParameterTokens)
+{
+    UartParameters parameters;
+    setManualBaudFlag();//STM32f429 does not support auto BaudRate
+
+    parseUartParameters(pParameterTokens, &parameters);
+    saveUartToBeUsedByDebugger(parameters.uartIndex);
+    if (parameters.share)
+        setUartSharedFlag();
+    else 
+	{
+        configureUartForExclusiveUseOfDebugger(&parameters);
+    }
+}
+
+
+static void parseUartParameters(Token* pParameterTokens, UartParameters* pParameters)
+{
+    memset(pParameters, 0, sizeof(*pParameters));
+    pParameters->uartIndex = 1;
+    pParameters->baudRate = 115200;
+
+    if (Token_MatchingString(pParameterTokens, "MRI_UART_MBED_USB"))

```

Comment:
You can remove this one if it doesn't make any sense for your platform.  I have it for the mbed-LPC1768 device since it has the USB interface chip connected to that UART and the mbed libs refer to it as USB_TX/USB_RX.